### PR TITLE
Apply dart code metrics suggestions

### PR DESCRIPTION
**Pull request by CHOOSEIT**
This small PR apply dart code metrics suggestions.

### CODE REVIEW

**Comment by CHOOSEIT** [comment]
Note that I haven't added the following suggestion: 


🎨 Prefer conditional expression.
in file lib/services/database/post_upvote_service.dart:51:5
prefer-conditional-expressions : https://dcm.dev/docs/rules/common/prefer-conditional-expressions


As I feel like it would make the code less readable: 
```dart
return voteState.exists
      ? VoteFirestore.fromDbData(voteState.data()!).hasUpvoted
        ? UpvoteState.upvoted
        : UpvoteState.downvoted
      : UpvoteState.none;```

File: lib/views/pages/login/login_page.dart [hunk 1]
```diff
@@ -18,18 +18,18 @@ class LoginPage extends HookConsumerWidget {
   @override
   Widget build(BuildContext context, WidgetRef ref) {
     final userRepository = ref.watch(userRepositoryProvider);
-    ref.listen(uidProvider, (_, user) async {
+    ref.listen(uidProvider, (_, user) {
```
Comment (camillelnne) [inline lib/views/pages/login/login_page.dart:21]: By curiosity, why is it synced now?

**Review by camillelnne — APPROVED** [review]
LGTM!
This was very satisfying to review, thanks for applying this refactor :)

**Review by TheTexanCodeur — APPROVED** [review]
LGTM!

Thanks for your work. The code looks even better now :)

**Review by yoannLafore — APPROVED** [review]
Thank you for applying the dart metrics recommendations :+1: 
This will improve the readibilty of our codebase.

The PR looks good to me.
Good job :tada:

**Pull request merged**